ReSharper.FSharp/src/FSharp.Common/src/Shim/TypeProviders/ExtensionTypingProviderShim.fs

```diff
@@ -26,8 +26,9 @@ type ExtensionTypingProviderShim(solution: ISolution, toolset: ISolutionToolset,
     let lifetime = solution.GetLifetime()
     let defaultShim = ExtensionTypingProvider
     let outOfProcess = experimentalFeatures.OutOfProcessTypeProviders
+    let createProcessLockObj = obj()
 
-    let mutable connection: TypeProvidersConnection = null
+    let [<VolatileField>] mutable connection: TypeProvidersConnection = null

```

Comment:
Is there any evidence this change is needed?

Reply:
"Does this need to be volatile? Best answer is Yes." (c)

The common implementation of double-checking locking requires a volatile attribute on the field being initialized, especially if the code works on different architectures.

Reply:
Shouldn't it be added to Fantomas host connection too? What about other connections in the product?